lib/trino-orc/src/main/java/io/trino/orc/reader/TimestampColumnReader.java

```diff
@@ -616,6 +617,11 @@ private void readInstantMicros(int i, long[] millisValues, int[] picosFractionVa
 
             // round nanos to micros and convert to picos
             picosFraction = toIntExact(roundDiv(nanos, NANOSECONDS_PER_MICROSECOND)) * PICOSECONDS_PER_MICROSECOND;
+
+            if (picosFraction == PICOSECONDS_PER_MILLISECOND) {
+                picosFraction = 0;
+                millis++;
+            }
```

Comment:
is this the only place like this in ORC time/timestamp readers?

Reply:
Would readInstantNanos have the same problem? I guess it doesn't include the rounding code, so it's probably ok.

Reply:
That method is safe, since it doesn't need to round.